src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -0,0 +1,122 @@
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                bool completedMessage;
+                WebSocketReceiveResult receiveResult;
+                ArraySegment<byte> buffer;
+                do
+                {
+                    buffer = new ArraySegment<byte>(new byte[1024]);
+                    receiveResult = await _webSocket.ReceiveAsync(buffer, cancellationToken);
+                    completedMessage = receiveResult.EndOfMessage;
+                    incomingMessage.Add(buffer.Array);
+                    totalBytes += buffer.Count;
+                } while (!completedMessage);
+
+                Message message;
+                if (incomingMessage.Count > 1)
+                {
+                    var totalBuffer = new byte[totalBytes];
+                    var offset = 0;
+                    for (int i = 0 ; i < incomingMessage.Count; i++)
+                    {
+                        System.Buffer.BlockCopy(incomingMessage[i], 0, totalBuffer, offset, incomingMessage[i].Length);
+                        offset += incomingMessage[i].Length;
+                    }
+
+                    message = new Message(ReadableBuffer.Create(totalBuffer).Preserve(), Format.Text, receiveResult.EndOfMessage);
+                }
+                else
+                {
+                    message = new Message(ReadableBuffer.Create(buffer.Array).Preserve(), Format.Text, receiveResult.EndOfMessage);
+                }
+
+                while (await _application.Output.WaitToWriteAsync(cancellationToken))
+                {
+                    if (_application.Output.TryWrite(message))
+                    {
+                        break;
+                    }
+                }
+            }
+        }
+
+        private async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
+        {
+            while (await _application.Input.WaitToReadAsync(cancellationToken))
+            {
+                Message message;
+                while (!cancellationToken.IsCancellationRequested && _application.Input.TryRead(out message))
+                {
+                    using (message)
+                    {
+                        await _webSocket.SendAsync(new ArraySegment<byte>(message.Payload.Buffer.ToArray()),
+                            message.MessageFormat == Format.Text ? WebSocketMessageType.Text : WebSocketMessageType.Binary, true,
+                            cancellationToken);
+                    }
+                }
+            }
+        }
+
+        public async Task Connect(string url)
+        {
+            await Connect(url, "");
+        }
+
+        public async Task Connect(string url, string queryString)
+        {
+            var wsUrl = url.Replace("http", "ws").Replace("https", "ws");
+            await _webSocket.ConnectAsync(new Uri(wsUrl + queryString), _cancellationToken);
+        }
+
+        public void Dispose()
```

Comment:
You will need to add StopAsync() - I am going to merge my PR soon. https://github.com/aspnet/SignalR/pull/183/files#diff-2e320ab431f1fe2e1ddea4ab7b1aebfbR12